core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnUniqueness.java

```diff
@@ -148,6 +148,13 @@ public Boolean areColumnsUnique(Intersect rel, RelMetadataQuery mq,
 
   public @Nullable Boolean areColumnsUnique(Sort rel, RelMetadataQuery mq,
       ImmutableBitSet columns, boolean ignoreNulls) {
+    // when return rows is less than or equals 1 (limit 1 or limit 0).
+    if (rel.fetch != null) {
+      int limit = RexLiteral.intValue(rel.fetch);
```

Comment:
Fetch and offset are not guaranteed to be RexLiteral. If you look at EnumerableSortRule you can see how that handles variables.

Reply:
Thank you very much. I will check this.

Reply:
done.